deepchem/utils/openvino_model.py

```diff
@@ -183,34 +208,74 @@ def __call__(self, generator: Iterable[Tuple[Any, Any, Any]]):
-
+  
     for inp_id, batch in enumerate(generator_copy):
       inputs, labels, weights = batch
       if self._keras_model is not None:
         self._keras_model._create_inputs(inputs)
         inputs, _, _ = self._keras_model._prepare_batch((inputs, None, None))
       elif self._torch_model is not None:
         inputs, _, _ = self._torch_model._prepare_batch((inputs, None, None))
-      inputs = inputs[0]
-
-      # Last batch size may be less or equal than overall batch size.
-      # Pad extra values by zeros and cut at the end.
-      last_batch_size = inputs.shape[0]
-      if last_batch_size != self._batch_size:
-        assert (last_batch_size < self._batch_size)
-        inp = np.zeros(
-            [self._batch_size] + list(inputs.shape[1:]), dtype=np.float32)
-        inp[:last_batch_size] = inputs
-        inputs = inp
+
+      inp_dict = {}
+      if(type(self._torch_model) == CGCNNModel):
+        self._torch_model.gcgnn.graph = inputs
+
+        node_feats = inputs.ndata.pop('x')
+        edge_feats = inputs.edata.pop('edge_attr')
+        inputs = [node_feats, edge_feats]
+
+        # Last batch size may be less or equal than overall batch size.
+        # Pad extra values by zeros and cut at the end.
+        last_batch_size = 20 % self._batch_size
+        last_batch_size = last_batch_size if last_batch_size is not 0 else 20
+
+        last_batch_sizes = [inputs[0].shape[0], inputs[1].shape[0]]
+        batch_sizes = [self._batch_size * 5, self._batch_size * 60]
+
+        for i in range(len(inputs)):
+          if last_batch_sizes[i] != batch_sizes[i]:
+            assert (last_batch_sizes[i] < batch_sizes[i])
+            inp = np.zeros(
+                [batch_sizes[i]] + list(inputs[i].shape[1:]), dtype=np.float32)
+            inp[:last_batch_sizes[i]] = inputs[i]
+            inputs[i] = inp
+
+        inp_names = list(self._exec_net.input_info.keys())
+        inp_name_len = len(inp_names)
+        inputs_len = len(inputs)
+
+        if inp_name_len != inputs_len:
+          sys.exit('Error: number of input names is ' \
+                  + str(inp_name_len) + ', but number of inputs is ' + str(inputs_len))
+
+        for i in range(len(inp_names)):
+          inp_dict[inp_names[i]] = inputs[i]
+      else:
+        assert (len(self._exec_net.input_info) == 1), 'Not implemented'
+        inputs = inputs[0]
+
+        # Last batch size may be less or equal than overall batch size.
+        # Pad extra values by zeros and cut at the end.
+        last_batch_size = inputs.shape[0]
+        
+        if last_batch_size != self._batch_size:
+          assert (last_batch_size < self._batch_size)
+          inp = np.zeros(
+              [self._batch_size] + list(inputs.shape[1:]), dtype=np.float32)
+          inp[:last_batch_size] = inputs
+          inputs = inp
+
+        inp_name = next(iter(self._exec_net.input_info.keys()))
+        inp_dict[inp_name] = inputs
```

Comment:
Let's try to unite both branches. Create fields to keep input and output shapes which are initialized at first call. Then we pad inputs by zeros in case of dimensions mismatch.